stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/messageoptions/MessageOptionItemVisibility.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2014-2024 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.ui.components.messageoptions
+
+import io.getstream.chat.android.compose.ui.components.selectedmessage.SelectedMessageMenu
+
+/**
+ * Controls menu item visibility in the selected message options menu. All options are visible by default.
+ *
+ * @param isRetryMessageVisible Visibility of the retry failed message option.
+ * @param isReplyVisible Visibility of the reply to message option (quote message).
+ * @param isThreadReplyVisible Visibility of the reply to message in thread option.
+ * @param isMarkAsUnreadVisible Visibility of the mark message as unread option.
+ * @param isCopyTextVisible Visibility of the copy message text option.
+ * @param isEditMessageVisible Visibility of the edit message option.
+ * @param isFlagMessageVisible Visibility of the flag message option.
+ * @param isPinMessageVisible Visibility of the pin message to chat option.
+ * @param isDeleteMessageVisible Visibility of the delete message option.
+ *
+ * @see [SelectedMessageMenu]
+ * @see [MessageOptions]
+ * @see [defaultMessageOptionsState]
+ */
+public data class MessageOptionItemVisibility(
+    val isRetryMessageVisible: Boolean = true,
+    val isReplyVisible: Boolean = true,
+    val isThreadReplyVisible: Boolean = true,
+    val isMarkAsUnreadVisible: Boolean = true,
+    val isCopyTextVisible: Boolean = true,
+    val isEditMessageVisible: Boolean = true,
+    val isFlagMessageVisible: Boolean = true,
+    val isPinMessageVisible: Boolean = true,
+    val isDeleteMessageVisible: Boolean = true,
```

Comment:
Are the property names ok or should they be more similar to the equivalent ones in the XML module? I guess the same client isn't using both XML and Compose so it shouldn't matter.

Reply:
Recently we migrated some XML component names to match the new naming we were providing to our component in Compose. In general, we should have similar names for both SKDs, It helps us in the future when we need to support our customers, doesn't matter which SDK they are using, the concept is the same.